### DISCUSSION TITLE
No Listing generated

### DISCUSSION THREAD

**Discussion by michiexile** [quarto-dev/quarto-cli#4266] [Q&A]
I'm writing a sequence of slides using Quarto and revealjs, and want to include an index file for my slides directory - both to be able to link to it, but also because the preview defaults to `index.html`, and I want more comfortable access to each of my slides.

So my directory contains
```
index.qmd
_quarto.yml
_definitions.qmd
extra.scss
L1.qmd
L2.qmd
L3.qmd
...
```

(as well as old cruft, including a random PDF, an `.rproj` file, an `.Rhistory` file, etc)

* `_definitions.qmd` - contains a few LaTeX `\def`, and gets included early in each of the lecture files.
* `L1.qmd`, ... each starts with something like
```
---
title: "Lecture 5"
format: revealjs
---

# Point Estimation

{{< include _definitions.qmd >}}
```
* `quarto.yml`
```
project:
  title: "Lectures MTH410"
  render:
    - "*.qmd"
    - "![FS][:digit:]{2}-lectures/"

execute:
  freeze: auto

author: Mikael Vejdemo-Johansson

format:
  revealjs:
    css: extra.css
    scrollable: true
    slide-number: c/t
    progress: true
    navigation-mode: linear
    code-fold: true
    error: false
    warning: false
    smaller: true
```
* `index.qmd`
```
---
title: "Lecture Slides and Notes"
format: html
listing:
  - id: lectures
    type: table
    contents:
      - "L[:digit:]+.qmd"
---

Here the lecture slides will appear as they are written.

::: {#lectures}
:::
```

No matter what I do, I cannot seem to get `index.qmd` to _actually_ list and link to the lecture slides themselves. `index.qmd` remains an empty page, except for the page title and whatever text I add before the div I provide for the listing itself. It doesn't matter whether I run `quarto render`, or use the Cmd-K shortcut in RStudio - same lack of results.

I don't understand why the listing does not generate. I don't understand how to get more debugging information out of quarto - trying to run preview or render with logging enabled does not give me any additional information.

**Comment by dragonstyle** [comment]
I just tinkered with a local version of this (not quite the same, but the basic structure) and I see the same issue when I use that global string that you are using (which _should_ work). When I use a very simple glob like `L*.qmd` it begins working. I'll take a closer look to see if I can get to the bottom of why that glob isn't working!

**Reply by michiexile** [reply]
I did try along the way with 

```
listing: table
```

```
listing:
  - type: table
    contents: "L*.qmd"
```

and several other setups with simple or no globs. I'm happy to switch out the glob if I can just get my listing to work that way.

**Reply by dragonstyle** [reply]
Hmm... The attached sample is working for me when I test locally (using the `L*.qmd` syntax). Does this work for you?

[sample.zip](https://github.com/quarto-dev/quarto-cli/files/10680827/sample.zip)

**Reply by dragonstyle** [reply]
(I should note I'm using the very latest 1.3 pre-release version, so if it isn't working for you, this could account for the difference)

**Comment by michiexile** [comment]
Your example worked well for me, whereas in my own setup I have now even tried listing the files one by one in my metadata with no success: 


```
---
title: "Lecture Slides and Notes"
format: html
listing:
  - id: lectures
    type: table
    contents:
      - L1.qmd
      - L2.qmd
      - L3.qmd
      - L4.qmd
      - L5.qmd
---

Here the lecture slides will appear as they are written.

::: {#lectures}
:::
```

...which is one reason I am so interested in finding out how to troubleshoot or extract diagnostics from quarto's process. Is there some way of telling _when_ in the processing things go wrong? The only printouts I've managed to get from the CLI indicate that it finds the right definitions of what I intend to include in the listing.

All the best,
--
Mikael Vejdemo-Johansson
***@***.*** ***@***.***>

**Comment by dragonstyle** [comment]
There is some information output when log-level is set to debug:

`quarto render index.qmd --log-level debug`

you should see some output prefixed with `[listing]` which should at least identify how the glob is being resolved. You may have already discovered this.

If there is some way for you to share the repo that is causing this (or email me a zip of the directory), I'd be happy to give it a try locally and see if I can see what is happening.

**Comment by michiexile** [comment]
The attached zipfile contains a somewhat peeled down version of the problem files (I made sure to only include the `.qmd` and the `_quarto.yml` on the off-chance that any of my cruft from previous years was interfering)

Here is also the entirety of `quarto render index.qmd --log-level debug`. Nothing with `[listing]` prefix shows up.

```
➜  quarto2 git:(S23) ✗ quarto render index.qmd --log-level debug
Error loading deno-dom-native: Could not open library: Could not open library: dlopen(/Applications/quarto/bin/tools/deno_dom/libplugin.dylib, 0x0005): tried: '/Applications/quarto/bin/tools/deno_dom/libplugin.dylib' (mach-o file, but is an incompatible architecture (have (x86_64), need (arm64e)))
pandoc --verbose --trace
  to: html
  output-file: index.html
  standalone: true
  section-divs: true
  html-math-method: mathjax
  wrap: none
  default-image-extension: png
  
metadata
  document-css: false
  link-citations: true
  date-format: long
  lang: en
  author: Mikael Vejdemo-Johansson
  title: Lecture Slides and Notes
  listing:
    - id: lectures
      type: table
      contents:
        - '*.qmd'
  
[INFO] Loaded /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/template.patched from /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/template.patched
[INFO] Loaded /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/metadata.html from /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/metadata.html
[INFO] Loaded /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/title-block.html from /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/title-block.html
[INFO] Loaded /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/title-metadata.html from /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/title-metadata.html
[INFO] Loaded /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/toc.html from /var/folders/rc/6mn6xqgd1694fxxlr6qfznbc0000gq/T/quarto-sessiond3ee30c7/5ab81106/e9046a5e/toc.html
[trace] Parsed [Plain [Str "Mikael",Space,Str "Vejdemo-Johansson"]] at line 1
[trace] Parsed [Plain [Str "Mikael",Space,Str "Vejdemo-Johansson"]] at line 1
[trace] Parsed [Plain [Str "long"]] at line 1
[trace] Parsed [Plain [Str "en"]] at line 1
[trace] Parsed [Plain [Str "*.qmd"]] at line 1
[trace] Parsed [Plain [Str "lectures"]] at line 1
[trace] Parsed [Plain [Str "table"]] at line 1
[trace] Parsed [Plain [Str "true"]] at line 1
[trace] Parsed [Plain [Str "1.2.313"]] at line 1
[trace] Parsed [Plain [Str "Lecture",Space,Str "Slides",Space,Str "and",Spa at line 1
[trace] Parsed [Plain [Str "Table",Space,Str "of",Space,Str "contents"]] at line 1
[trace] Parsed [] at line 5
[trace] Parsed [Para [Str "Here",Space,Str "the",Space,Str "lecture",Space, at line 7
[trace] Parsed [Div ("lectures",[],[]) []] at line 11
[INFO] Running filter /Applications/quarto/share/filters/quarto-init/quarto-init.lua
[INFO] Completed filter /Applications/quarto/share/filters/quarto-init/quarto-init.lua in 17 ms
[INFO] Running filter /Applications/quarto/share/filters/authors/authors.lua
[INFO] Completed filter /Applications/quarto/share/filters/authors/authors.lua in 33 ms
[INFO] Running filter /Applications/quarto/share/filters/quarto-pre/quarto-pre.lua
[INFO] Completed filter /Applications/quarto/share/filters/quarto-pre/quarto-pre.lua in 12 ms
[INFO] Running filter /Applications/quarto/share/filters/crossref/crossref.lua
[INFO] Completed filter /Applications/quarto/share/filters/crossref/crossref.lua in 9 ms
[INFO] Running filter /Applications/quarto/share/filters/layout/layout.lua
[INFO] Completed filter /Applications/quarto/share/filters/layout/layout.lua in 10 ms
[INFO] Running filter /Applications/quarto/share/filters/quarto-post/quarto-post.lua
[INFO] Completed filter /Applications/quarto/share/filters/quarto-post/quarto-post.lua in 7 ms
[INFO] Running filter /Applications/quarto/share/filters/quarto-finalize/quarto-finalize.lua
[INFO] Completed filter /Applications/quarto/share/filters/quarto-finalize/quarto-finalize.lua in 7 ms
Output created: index.html

➜  quarto2 git:(S23) ✗ 
```

I am noticing a failure to load `deno_dom` in the beginning. I am working on an Apple Silicon M1 MacBook Pro. Do I need to build quarto myself to be able to use it?

All the best,
--
Mikael Vejdemo-Johansson
***@***.***

**Reply by dragonstyle** [reply]
I don't think the attachment made it through email to Github - can you try uploading it directly. Two other things as we debug:

- What version of Quarto are you using? The fact the you have no listing output likely means we're on very different versions...

- The deno_dom warning shouldn't affect any functionality - that is just there so we know we're falling back to a WASM version of that library since we don't have a statically compiled one loaded for your architecture, shouldn't affect anything.

**Comment by michiexile** [comment]
[quartoproblem.zip](https://github.com/quarto-dev/quarto-cli/files/10686824/quartoproblem.zip)

I'm using the Quarto that Homebrew gave me, version 1.2.313.

**Reply by dragonstyle** [reply]
Ok this explained it! The project type for your sample is not a website and listings are only support for website projects. Updating the `_quarto.yml` to:

```yml

project:
  title: "Lectures MTH410"
  render:
    - "*.qmd"
    - "![FS][:digit:]{2}-lectures/"
  type: website # < This is the new new line

execute:
  freeze: auto

author: Mikael Vejdemo-Johansson

format:
  revealjs:
    css: extra.css
    scrollable: true
    slide-number: c/t
    progress: true
    navigation-mode: linear
    code-fold: true
    error: false
    warning: false
    smaller: true
```

Updating the listing page to:

```markdown
---
title: "Lecture Slides and Notes"
format: html
listing:
  - id: lectures
    type: table
    contents:
      - "L[[:digit:]+].qmd"
---

Here the lecture slides will appear as they are written.

::: {#lectures}
:::
```

Got the glob resolving properly.

Issue #1204 is tracking a better error / warning for this case (or supporting listings outside of a website).

Sorry to have your time sink into something we should be warning you about!

**Reply by michiexile** [reply]
Excellent! Now it works!

Thank you very much for your help!

**Comment by michiexile** [comment]
I have also tried it (just now) with the same lack of success with version 1.2.335, from the latest release version from here.
